src/Exception/DomainException.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\Common\Exception;
+
+use DomainException as SplDomainException;
+use InvalidArgumentException;
+use Throwable;
+
+class DomainException extends SplDomainException
+{
+    public const DEFAULT_STATUS_CODE = 500;
+
+    /**
+     * @var string
+     */
+    private $token;
```

Comment:
Very few things are immutable. The cornerstone of software architecture is embracing change.